core/lib/multivm/src/versions/vm_fast/tests/tester/vm_tester.rs

```diff
@@ -69,9 +73,19 @@ impl VmTester {
             // `insert_contracts(&mut self.storage, &self.custom_contracts);`
         }
 
+        let storage = self.storage.clone();
+        {
+            let mut storage = storage.borrow_mut();
+            // Commit pending storage changes (old VM versions commit them on successful execution)
+            for (&(address, slot), &value) in self.vm.inner.world_diff.get_storage_state() {
+                let key = StorageKey::new(AccountTreeId::new(address), u256_to_h256(slot));
+                storage.set_value(key, u256_to_h256(value));
+            }
+        }
+
```

Comment:
Why exactly is this happening in the `reset_state`?

Reply:
yeah but this relies on the user of the tester to call the reset at appropriate times and only then checking expected values in storage, this can be easily misused?

Reply:
IIUC, it's not really the case, we generally use `WorldDiff`-aware methods to check storage slots, like [this one](https://github.com/matter-labs/zksync-era/blob/d66a4a49d454f0b5fed4cfb80898281b2711bc1d/core/lib/multivm/src/versions/vm_fast/tests/utils.rs#L26). It's just that specifically resetting a VM in existing tests implies that all storage changes are committed, which is difficult to reproduce without committing these changes (since the storage is accessed not by the test code, but by a subsequent VM run in this case).